### PR TITLE
Support parallel 64- and 32-bit boost binaries (RIPD-1317):

### PR DESCRIPTION
**Pull request by ximinez**
* CMake and scons
* Update Visual Studio build docs
* Cache Appveyor PIP downloads for scons builds
* Rename the CMakeFuncs `find_` functions to `get_`

### CODE REVIEW

**Comment by vinniefalco** [comment]
I don't understand. rippled can only be built in the 64 bit address model. What's the 32 bit for?

**Comment by ximinez** [comment]
It's to support the use case where the user has a 32-bit and 64-bit _boost_ build in the `stage` and `stage64` directories, respectively. The issue came up in the context of https://github.com/vinniefalco/Beast/pull/110, and I figured it didn't make sense to require different boost configurations for beast and rippled.

**Comment by ximinez** [comment]
Specifically, nobody is _required_ to have both 32-bit and 64-bit boost, but if they do for some other project, rippled will build correctly.

File: Builds/CMake/CMakeFuncs.cmake [hunk 1]
```diff
@@ -268,8 +268,21 @@ endmacro()
 ############################################################
 
 # Params: Boost components to search for.
-macro(find_boost)
-  if (NOT WIN32)
+macro(get_boost)
```
Comment (seelabs) [inline Builds/CMake/CMakeFuncs.cmake]: Nit: I prefer `use` to `get`. These macros are actually changing compiler search paths. `use_boost` sounds like it's going to do something. Both `find_boost` and `get_boost` sound like pure functions.

**Comment by seelabs** [comment]
:+1: The code looks good to me. I can't easily test on windows, but I'm assuming both you  and vinnie will test on that platform.

**Comment by vinniefalco** [comment]
All scons and CMake targets build correctly in VS2015!
:+1:

File: Builds/CMake/CMakeFuncs.cmake (hunk 1 above)
Comment (ximinez) [inline Builds/CMake/CMakeFuncs.cmake]: Sounds good. Will change.

**Comment by ximinez** [comment]
Changed the `get_` functions to `use_`. As soon as they pass CI (and I can't imagine why they wouldn't), I'll squash and update the commit message.

**Comment by nbougalis** [comment]
Merged as fd901f8081bdcd50c4027e29738d579c8727821f

**Pull request closed without merging**